src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1649,32 +1654,85 @@ public function getLanguages(): array
         $this->languages = [];
         foreach ($languages as $acceptHeaderItem) {
             $lang = $acceptHeaderItem->getValue();
-            if (str_contains($lang, '-')) {
-                $codes = explode('-', $lang);
-                if ('i' === $codes[0]) {
-                    // Language not listed in ISO 639 that are not variants
-                    // of any listed language, which can be registered with the
-                    // i-prefix, such as i-cherokee
-                    if (\count($codes) > 1) {
-                        $lang = $codes[1];
-                    }
-                } else {
-                    for ($i = 0, $max = \count($codes); $i < $max; ++$i) {
-                        if (0 === $i) {
-                            $lang = strtolower($codes[0]);
-                        } else {
-                            $lang .= '_'.strtoupper($codes[$i]);
-                        }
-                    }
-                }
-            }
-
-            $this->languages[] = $lang;
+            $this->languages[] = $this->formatLocale($lang);
         }
+        $this->languages = array_unique($this->languages);
 
         return $this->languages;
     }
 
+    /**
+     * This method strips the locale to only keep the canonicalized language value.
```

Comment:
blank line after (to separate the title from the description)

```suggestion
     * Strips the locale to only keep the canonicalized language value.
     *
```